dummyfs/dummyfs.c

```diff
@@ -14,65 +14,69 @@
 
 #include <stdio.h>
 #include <stdint.h>
-#include <stdlib.h>
 #include <stdarg.h>
 #include <sys/stat.h> /* to set mode for / */
-#include <sys/threads.h>
-#include <sys/msg.h>
 #include <sys/list.h>
-#include <sys/mman.h>
 #include <sys/mount.h>
+#include <sys/threads.h>
 #include <string.h>
-#include <unistd.h>
+#include <stdlib.h>
 #include <dirent.h>
-#include <signal.h>
 #include <poll.h>
-#include <phoenix/sysinfo.h>
 
 #include "dummyfs.h"
 #include "dir.h"
 #include "file.h"
 #include "object.h"
 #include "dev.h"
 
-#define LOG(msg, ...) printf("dummyfs: " msg, ##__VA_ARGS__)
 
-struct _dummyfs_common_t dummyfs_common;
+int dummyfs_incsz(dummyfs_t *ctx, int size)
+{
+	if (ctx->size + size > DUMMYFS_SIZE_MAX)
+		return -ENOMEM;
+	ctx->size += size;
+	return EOK;
+}
 
-int dummyfs_destroy(oid_t *oid);
+void dummyfs_decsz(dummyfs_t *ctx, int size)
+{
+	ctx->size -= size;
+}
 
 
-static inline int dummyfs_device(oid_t *oid)
+static inline int dummyfs_device(dummyfs_t *ctx, oid_t *oid)
 {
-	return oid->port != dummyfs_common.port;
+	return oid->port != ctx->port;
 }
 
 
-static inline dummyfs_object_t *dummyfs_get(oid_t *oid)
+static inline dummyfs_object_t *dummyfs_get(dummyfs_t *ctx, oid_t *oid)
 {
-	return dummyfs_device(oid) ? dev_find(oid, 0) : object_get(oid->id);
+	return dummyfs_device(ctx, oid) ? dev_find(ctx, oid, 0) : object_get(ctx, oid->id);
 }
 
 
-int dummyfs_lookup(oid_t *dir, const char *name, oid_t *res, oid_t *dev)
+int dummyfs_lookup(void *ctx, oid_t *dir, const char *name, oid_t *res, oid_t *dev)
 {
+	dummyfs_t *dummyfs = (dummyfs_t *)ctx;
```

Comment:
I'm somewhat worried about this variable name, there is some chance it will cover some symbol. Maybe `fsctx` or something?

Reply:
Applies also to other instances of such name